Source/ReactiveCore/MoyaError.swift

```diff
@@ -9,3 +9,59 @@ public enum MoyaErrorCode: Int {
     case StatusCode
     case Data
 }
+
+public enum MoyaError: ErrorType {
+    case ImageMapping(MoyaResponse)
+    case JSONMapping(MoyaResponse)
+    case StringMapping(MoyaResponse)
+    case StatusCode(Int,MoyaResponse)
+    case Data(MoyaResponse)
+    case Underlying(ErrorType)
+    
+    public func sameErrorType(otherError: MoyaError) -> Bool {
```

Comment:
This isn't equality per se though; to my mind that would involve the associated values matching as well.

Moving it to an extension makes sense; I'm trying to come up with non-testing use cases but nothing so far. Maybe we move it into the test target for now and see if we miss it?